docs/guides/color.md

```diff
@@ -0,0 +1,290 @@
+- BOP (blue, orange, purple) colormap names: `'bop blue', 'bop orange', 'bop purple'`.
+
+More specifically, the available colormap names can be listed at any time.
+
+```python
+In [8]: list(napari.utils.colormaps.AVAILABLE_COLORMAPS)
+Out[8]:
+['PiYG',
+ 'blue',
+ 'bop blue',
+ ...,
+ 'viridis',
+ 'yellow']
+```
+
+If there is a need for a custom colormap that is not available by default,
+pass an instance of a napari `Colormap`
+
+
+```python
+image.colormap = Colormap(
+    colors=[[0, 0, 0, 1], [0, 0, 1, 1]],
+    name='blue',
+    interpolation='linear',
+    controls=[0, 1],
+)
+```
+
+which will be added to `AVAILABLE_COLORMAPS` after it is successfully created.
+Equivalently, pass a `dict` representation of the same colormap.
+
+```python
+image.colormap = {
+    'colors': [[0, 0, 0, 1], [0, 0, 1, 1]],
+    'name': 'blue',
+    'interpolation': 'linear',
+    'controls': [0, 1],
+}
+```
+
+Any single color value that is not a string can also be used
+
+```python
+image.colormap = (0, 0, 1)
+```
+
+in which case a colormap from black to the given color will be generated.
+In general, the color value cannot be given as a name string because strings
+are reserved to refer to existing colormap names.
+
+
+### Categorical maps
+
+The [`CategoricalColormap`](https://github.com/napari/napari/blob/c86cf87a788b1bbe62afc0b92b9ebdca1331a3e5/napari/utils/colormaps/categorical_colormap.py#L13)
+class defines a map from a categorical or discrete value to a color.
+It is defined by two components.
+
+1. A dictionary that maps from values to colors.
+2. A cycle of fallback colors that used when a value is not found in the dictionary.
+
+For example, a points layer's face color could be mapped from an optional cell type
+
+```python
+In  [9]: points.properties = {'cell_type': ['astrocyte', 'microglia', None]}
+In [10]: points.face_color_cycle = {
+             'colormap': {
+                 'astrocyte': 'red',
+                 'oligodendrocyte': 'yellow',
+                 None: 'blue',
+             },
+             'fallback_color': ['lime', 'magenta', 'cyan'],
+         }
+In [11]: points.face_color = 'cell_type'
+In [12]: points.face_color
+Out[12]:
+array([[1., 0., 0., 1.],
+       [0., 1., 0., 1.],
+       [0., 0., 1., 1.]], dtype=float32)
+```
+
+Alternatively, a dictionary alone can be used to define the mapping
+
+```python
+In [13]: points.face_color_cycle = {
+            'astrocyte': 'red',
+            'oligodendrocyte': 'yellow',
+            None: 'blue',
+         }
+```
+
+in which case the fallback color will default to `['white']`.
+
+Or a sequence alone can be used to defined the fallback color cycle
+
+```python
+In [14]: points.face_color_cycle = ['lime', 'magenta', 'cyan']
+```
+
+in which case the mapping will initially be an empty dictionary.
+As values from the fallback cycle are used, the value to color mappings
+they define will be added to the dictionary.
+
```

Comment:
Should there be a section on how to make custom colormaps?

Reply:
Good point. For the `CategoricalColormap`, it's effectively always custom because there are no default/built-in ones (though maybe there should be!).

I added a bit more detail and some extra subsections to help clarify how to customize a colormap. Please take another look after the next push.

Reply:
Should we warn users that when using a cycle then the color mapping may not be stable between ?
I'm not too familiar with `ColorCycle`, can they be infinite when creating them manually instead of base on a dict ? Like i can imagine writing a generator that always return a new color ?

Reply:
By not stable, I assume you mean that as input values are found that are missing from the dict, we use the next value from the fallback cycle (which does repeat indefinitely), and add a new entry to the dict (i.e. the dict is mutated as missing values are found). And that this is sensitive to the input data - specifically the order in which the input values are found.
If so, I added an extra to explicitly clarify that's the behavior here.

Reply:
Yeah seeing it written out is very useful but makes me feel like this is a bad model and we should not support it going forward. That is, maybe categorical colormaps should be required to be a direct *and complete* mapping?

Of course, this is the current behaviour so I'm cool with leaving it here and documented and updating the docs when we change it, I just mean, seeing it written out really makes me think that it's pretty absurd.

And just to go back full circle, I can see as a user how you might *not know* ahead of time what the categories are, and in that case you really just want to say "map this to a bunch of different colours for me thx". Perhaps some utility functions to create categorical maps from data can be exposed?

Anyway, if we all agree, I guess we can make a specific issue about it, but wanted to get everyone's feeling on this.

Reply:
Are you maybe annoyed by the fact that the undetermined categories are a cycle ? 
Would you feel better if the reasoning to choose the color is an implementation detail of the colormap ? Like a hash of the category that could be stable ? 

But maybe let's move that a a separate issue ?

Reply:
I think it's OK to have a little discussion here, because docs are a good way to clarify behavior and generate discussion.

Agreed there are two main use cases here.

1. You know all the possible discrete values and want to completely specify the mapping. The single default fallback color is useful 
2. You don't know all the possible discrete values and/or don't care how they are mapped, but want to pick the set of colors they are mapped to.

I think the current API and implementation of `CategoricalColormap` fulfills both use cases, so I think it's mostly fine as is. In use case 2 the `CategoricalColormap::map` has side-effects on the mapping that are sensitive to order of data encountered, but in that use-case we don't really care about the details of the mapping (except the range/values), so I don't think it's too important.

It also fulfills a third use case, which mixes the two - you specify a partial mapping and more than one fallback color - which I doubt is common, but I don't think there's much harm in supporting it.